base/libgit2/types.jl

```diff
@@ -177,22 +177,44 @@ function RemoteCallbacks{P<:AbstractPayload}(credentials::Ptr{Void}, payload::Nu
     end
 end
 
-immutable FetchOptions
-    version::Cuint
-    callbacks::RemoteCallbacks
-    prune::Cint
-    update_fetchhead::Cint
-    download_tags::Cint
-end
-FetchOptions(; callbacks::RemoteCallbacks = RemoteCallbacks(),
-               prune::Cint = Consts.FETCH_PRUNE_UNSPECIFIED,
-               update_fetchhead::Cint = one(Cint),
-               download_tags::Cint = Consts.REMOTE_DOWNLOAD_TAGS_AUTO
-) = FetchOptions(one(Cuint),
-                 callbacks,
-                 prune,
-                 update_fetchhead,
-                 download_tags)
+if LibGit2.version() >= v"0.24.0"
+    immutable FetchOptions
+        version::Cuint
+        callbacks::RemoteCallbacks
+        prune::Cint
+        update_fetchhead::Cint
+        download_tags::Cint
+        custom_headers::StrArrayStruct
+    end
+    FetchOptions(; callbacks::RemoteCallbacks = RemoteCallbacks(),
+                   prune::Cint = Consts.FETCH_PRUNE_UNSPECIFIED,
+                   update_fetchhead::Cint = one(Cint),
+                   download_tags::Cint = Consts.REMOTE_DOWNLOAD_TAGS_AUTO,
+                   custom_headers::StrArrayStruct = StrArrayStruct()
+    ) = FetchOptions(one(Cuint),
```

Comment:
The whole the file uses this convention... I've added a commit fixing this everywhere.